model/policy.go

```diff
@@ -157,15 +158,19 @@ func (model Model) AddPolicy(sec string, ptype string, rule []string) {
 				}
 				if idx > idxInsert {
 					assertion.Policy[i] = assertion.Policy[i-1]
+					assertion.PolicyMap[strings.Join(assertion.Policy[i-1], DefaultSep)]++
 				} else {
 					break
 				}
 			}
 			assertion.Policy[i] = rule
 			assertion.PolicyMap[strings.Join(rule, DefaultSep)] = i
+			isSet = true
```

Comment:
`isSet` only be set true at line 168, so why not return directly here?

Reply:
It'ok. Just move the line ``assertion.PolicyMap[strings.Join(rule, DefaultSep)] = len(model[sec][ptype].Policy) - 1`` up